Assets/Plugins/Source/Editor/Configs/EditorConfig.cs

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace PlayEveryWare.EpicOnlineServices.Editor.Config
+{
+    using System.Threading.Tasks;
+    using UnityEditor;
+    using UnityEngine;
+    using Config = EpicOnlineServices.Config;
+
+    /// <summary>
+    /// Used for configurations that are editor-only.
+    /// Configurations are serialized into JSON strings, and then added to EditorPrefs
+    /// using their filename as the key.
+    /// </summary>
+    public abstract class EditorConfig : Config
+    {
+        protected EditorConfig(string filename) : base(filename, string.Empty) { }
+
+        public override Task WriteAsync(bool prettyPrint = true)
+        {
+            // TODO: Add JSON schemas for validation.
+            string configAsJSON = JsonUtility.ToJson(this, prettyPrint);
+            EditorPrefs.SetString(Filename, configAsJSON);
+            return Task.CompletedTask;
+        }
+
+        protected override Task ReadAsync()
+        {
+            // TODO: Add JSON schemas for validation.
+            string configAsJSON = EditorPrefs.GetString(Filename);
+            JsonUtility.FromJsonOverwrite(configAsJSON, this);
+            return Task.CompletedTask;
```

Comment:
Take a look at "Shared preferences" described here: https://blog.redbluegames.com/how-to-handle-data-for-custom-editor-tools-in-unity-6b85e9e17715

Users will want to share their config values and have them stored in source control.

Reply:
Functionality was changed to write configuration values of an "editor" nature to `etc/config/`